rustls/src/quic.rs

```diff
@@ -519,6 +519,56 @@ impl Version {
 mod test {
     use super::*;
 
+    #[test]
+    fn short_packet_header_protection() {
+        // https://www.rfc-editor.org/rfc/rfc9001.html#name-chacha20-poly1305-short-hea
+
+        const PN: u64 = 654360564;
+        const SECRET: &[u8] = &[
+            0x9a, 0xc3, 0x12, 0xa7, 0xf8, 0x77, 0x46, 0x8e, 0xbe, 0x69, 0x42, 0x27, 0x48, 0xad,
+            0x00, 0xa1, 0x54, 0x43, 0xf1, 0x82, 0x03, 0xa0, 0x7d, 0x60, 0x60, 0xf6, 0x88, 0xf3,
+            0x0f, 0x21, 0x63, 0x2b,
+        ];
+
+        let secret = hkdf::Prk::new_less_safe(hkdf::HKDF_SHA256, SECRET);
+        use crate::tls13::TLS13_CHACHA20_POLY1305_SHA256_INTERNAL;
+        let hpk = HeaderProtectionKey::new(TLS13_CHACHA20_POLY1305_SHA256_INTERNAL, &secret);
+        let packet = PacketKey::new(TLS13_CHACHA20_POLY1305_SHA256_INTERNAL, &secret);
+
+        const PLAIN: &[u8] = &[0x42, 0x00, 0xbf, 0xf4, 0x01];
+
+        let mut buf = PLAIN.to_vec();
+        let (header, payload_tag) = buf.split_at_mut(4);
+        let tag = packet
+            .encrypt_in_place(PN, &*header, payload_tag)
```

Comment:
s/payload_tag/payload/; the tag isn't there yet

Reply:
Yup; fixed.